### PR TITLE
feat(init-1D): #MVS-128 reset initialization status on year transition

### PR DESCRIPTION
**Pull request by flmartineau**
## Describe your changes

- added reset of initialization status on the year transition event

## Checklist tests

- check if the transition event allow the init form to be displayed again (this can be tester through the API : https://confluence.support-ent.fr/pages/viewpage.action?pageId=67214467)

## Issue ticket number and link

https://jira.support-ent.fr/browse/MVS-128

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/viescolaire/service/impl/VieScolaireRepositoryEvents.java [hunk 1]
```diff
@@ -139,8 +140,14 @@ public void handle (Either<String, JsonObject> event) {
 
     }
 
+    @Override
     public void transition(JsonObject structure) {
-        log.info("[VieScolaireRepositoryEvents] : transition event is not implemented");
+        this.initService.setInitializationStatus(structure.getString(Field.ID), false)
+                .onFailure(fail -> {
+                    log.error("[VieScolaireRepositoryEvents] : An error occured when setting initialization status");
+                    log.error(fail.getMessage());
```
Comment (JinDouang) [inline src/main/java/fr/openent/viescolaire/service/impl/VieScolaireRepositoryEvents.java:148]: Dans le même log.error ?

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: src/main/java/fr/openent/viescolaire/service/impl/VieScolaireRepositoryEvents.java (hunk 1 above)
Comment (flmartineau) [reply src/main/java/fr/openent/viescolaire/service/impl/VieScolaireRepositoryEvents.java:148]: j'ai repris le format des autres logs pour pas que ca soit trop différent

Comment (flmartineau) [reply src/main/java/fr/openent/viescolaire/service/impl/VieScolaireRepositoryEvents.java:148]: tu préferes que je change ?

**Review by BastienBenard — APPROVED** [review]

File: src/main/java/fr/openent/viescolaire/service/impl/VieScolaireRepositoryEvents.java (hunk 1 above)
Comment (JinDouang) [reply src/main/java/fr/openent/viescolaire/service/impl/VieScolaireRepositoryEvents.java:148]: Tant qu'à faire... pour éviter d'avoir 2 lignes

**Review by simonchaval — APPROVED** [review]

**Pull request merged**